### ISSUE TITLE
dbhome confusion

### ISSUE THREAD

**Issue by stelb** [oravirt/ansible-oracle#122]
i tried to install two databases with different versions and got this error

TASK [oradb-manage-db : manage-db | create/manage database] ***************************
fatal: [db01n2.dev.int.nexinto.com]: FAILED! => {"changed": false, "msg": "Database TEST already exists in a different home. Stdout -> PRCD-1229 : An attempt to access configuration of database TEST was rejected because its version 12.1.0.2.0 differs from the program version 12.2.0.1.0. Instead run the program from /www/oracle/12.1.0.2/ee.\n"}

i removed the second database from my config and still the same error, it only succeeded after removal of the second dbhome.

ansible-oracle ff2604cc6767e24c395139e7a3f83e4bbcecc1da
modules: 23510bf00465ff8d508ee62b5d426463bb4ab47c

**Comment by oravirt** [comment]
So, you had 2 homes: 12.2.0.1 & 12.1.0.2 and the first database was created in the 12.1.0.2 home?
Is that correct?

**Comment by stelb** [comment]
here it is:
```
db_homes_config:
  ee-12.1.0.2:
     home: ee
     version: 12.1.0.2
     edition: EE
  se2-12.2.0.1:
     home: se2
     version: 12.2.0.1
     edition: SE2

db_homes_installed:
     - home: ee-12.1.0.2
       apply_patches: False
       state: present
     - home: se2-12.2.0.1
       apply_patches: False
       state: present
```

**Comment by stelb** [comment]
and yes, the first db created was 12.1.0.2

```
---
oracle_databases:
      - home: ee-12.1.0.2
        oracle_db_name: TEST
        oracle_db_type: SI
        is_container: True
        storage_type: ASM
        oracle_db_mem_totalmb: 4096
        oracle_database_type: MULTIPURPOSE
        redolog_size: 100M
        datafile_dest: '+SI_DATA'
        recoveryfile_dest: '+SI_DATA'
        archivelog: false
        flashback: false
        force_logging: false
        state: present
```

**Comment by oravirt** [comment]
Thanks, I'll try to re-produce this

**Comment by oravirt** [comment]
I have not seen any difference in behaviour in different Ansible versions, but:
This problem looks like it's a matter of scope, as in depending on which roles you have in your playbook.
In the `oraswdb-install` and `oraswdb-manage-patches` roles, I use `set_fact` to define the `oracle_home_db` variable to the ORACLE_HOME I'm working on for 'that' iteration of the loop. Later, when you get to the `oradb-manage-db` role that variable is still set (as there is no way to 'unset' a variable). 
By default, the `oradb-manage-db` role gets the value for `oracle_home_db`  from the `orasw-meta` role defaults. But, since the variable is already set from the previous role and has a higher priority the variable is never actually evaluated for each iteration of the `manage-db.yml` in `oradb-manage-db`. So, it will continue to be set to the last value it got from the previous role.

But, if the `oradb-manage-db` role is run in isolation, this problem never occurs and it works as expected - the variable is evaluated for each iteration and the variable gets set to its correct value for each iteration.

So the solution is to use `set_fact` in the `manage-db.yml` playbook as well. Changes will be pushed asap.

**Comment by oravirt** [comment]
Not sure how I could have missed this in testing, but now I know.....

Thanks for pointing this out!

**Comment by Rendanic** [comment]
I had some problems with Ansible 2.5 from EPEL some months ago. It was painful to find that problem, because set_fact wasn't working under special conditions and I found the bug with the exact description on the Ansible project on github.
That was really painful, because I hit the problem with an update of Ansible from EPEL. The solution was switching back to 2.4. That was the reason why I worked with 2.4 for a long time. Now I switched back to EPEL again and hit the next issue in 2.7.0 which was confirmed as well. 
The most current version is working without any issue.

We have to keep an eye on the used version of Ansible when trying to reproduce problems in playbooks. 'ansible-oracle' is quite a complex project with lots of variables and switches. It is not possible to test all the possible combinations...

But Mikael did a fantastic job in the project!
